ext/opentelemetry-ext-cloud-trace/src/opentelemetry/ext/cloud_trace/__init__.py

```diff
@@ -0,0 +1,266 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Cloud Trace Span Exporter for OpenTelemetry."""
+
+import logging
+from typing import Sequence, Dict, Any, List
+
+from google.cloud import trace_v2
+from google.cloud.trace_v2 import TraceServiceClient
+import opentelemetry.trace as trace_api
+from google.cloud.trace_v2.proto.trace_pb2 import AttributeValue
+from opentelemetry.context import Context
+from opentelemetry.sdk.trace import Event
+from opentelemetry.sdk.trace.export import Span, SpanExporter, SpanExportResult
+from opentelemetry.sdk.util import ns_to_iso_str
+from opentelemetry.util import types
+
+from opentelemetry.ext.cloud_trace.version import __version__
+
+logger = logging.getLogger(__name__)
+
+AGENT = "opentelemetry-python [{}]".format(__version__)
+# Max length is 128 bytes for a truncatable string.
+MAX_LENGTH = 128
+
+
+class CloudTraceSpanExporter(SpanExporter):
+    """Cloud Trace span exporter for OpenTelemetry.
+
+    Args:
+        client: Cloud Trace client.
+        project_id: project_id to create the Trace client.
+    """
+
+    def __init__(
+            self, client=None, project_id=None,
+    ):
+        if client is None:
+            client = TraceServiceClient()
+        self.client = client
+        self.project_id = project_id
```

Comment:
What happens if project_id is None?

Reply:
Was this resolved? A `None` project_id is likely to be the most common scenario. We may even want to default it to `None` and let the TraceServiceClient figure it out from the default credentials.

Reply:
Hey Marius. I made project_id a required argument since there doesn't seem to be a way to get project_id from the TraceServiceClient. This functionality seems to have removed going from trace_api v1 to v2.

Reply:
ok, perhaps we need to postpone this.
As a quickfix, you may be able to do `credentials, project_id = google.auth.default()`.